torchvision/prototype/models/_utils.py

```diff
@@ -1,13 +1,55 @@
+import functools
 import warnings
-from typing import Any, Dict, Optional, TypeVar
+from typing import Any, Dict, Optional, TypeVar, Callable, Tuple, Union
 
-from ._api import WeightsEnum
+from torch import nn
+from torchvision.prototype.utils._internal import kwonly_to_pos_or_kw
 
+from ._api import WeightsEnum
 
 W = TypeVar("W", bound=WeightsEnum)
+M = TypeVar("M", bound=nn.Module)
 V = TypeVar("V")
 
 
+def handle_legacy_interface(
+    **pretrained_weights: Tuple[str, Union[Optional[W], Callable[[Dict[str, Any]], Optional[W]]]]
+):
+    def outer_wrapper(builder: Callable[..., M]) -> Callable[..., M]:
+        @functools.wraps(builder)
+        def inner_wrapper(**kwargs: Any) -> M:
+            for weights_param, (pretrained_param, default) in pretrained_weights.items():  # type: ignore[union-attr]
+                weights_arg = kwargs.get(weights_param)
+                if weights_param in kwargs and not isinstance(weights_arg, bool):
```

Comment:
This correct way of handling it. You need to check again that `weights_param in kwargs` instead of doing `weights_param is not None` to handle properly all cases. But this can be unexpected by readers of the code so please add comments to explain why you need to handle it the way you do. My concern is that someone might refactor it without understanding the implications.